src/net.h

```diff
@@ -838,37 +838,89 @@ class CNode
         assert(false);
     }
 
+    /* These are the default connections that we use to connect with the
+     * network. There is no restriction on the inventory relayed- by default we
+     * relay blocks, addresses & transactions. We attempt to open
+     * MAX_OUTBOUND_FULL_RELAY_CONNECTIONS using addresses from our AddrMan.
+     */
     bool IsFullOutboundConn() const {
-        return m_conn_type == ConnectionType::OUTBOUND;
+        return m_conn_type == ConnectionType::OUTBOUND_FULL_RELAY;
     }
 
+    /* We open manual connections to addresses that users explicitly inputted
+     * via the addnode RPC, or the -connect command line argument. Even if a
+     * manual connection is misbehaving, we do not automatically disconnect or
+     * add it to our discouragement filter.
+     */
     bool IsManualConn() const {
         return m_conn_type == ConnectionType::MANUAL;
     }
 
+    /* After meeting our quota for full outbound connections, we attempt to
+     * open MAX_BLOCK_RELAY_ONLY_CONNECTIONS to help prevent against partition
+     * attacks. By not relaying transactions or addresses, these connections
+     * are harder to detect by a third party, thus helping obfuscate the
+     * network topology.
+     */
     bool IsBlockOnlyConn() const {
         return m_conn_type == ConnectionType::BLOCK_RELAY;
     }
 
+    /* Feeler connections are short lived connections used to increase the
+     * number of connectable addresses in our AddrMan. Approximately every
+     * FEELER_INTERVAL, we attempt to connect to a random address from the new
+     * table. If successful, we add it to the tried table.
+     */
     bool IsFeelerConn() const {
         return m_conn_type == ConnectionType::FEELER;
     }
 
+    /* AddrFetch connections are short lived connections used to solicit
+     * addresses from peers. These are initiated to addresses submitted via the
+     * -seednode command line argument, or under certain conditions when the
+     * AddrMan is empty.
+     */
     bool IsAddrFetchConn() const {
         return m_conn_type == ConnectionType::ADDR_FETCH;
     }
 
+    /* Inbound connections are those initiated by a peer. This is the only
+     * property we know at the time of connection, until P2P messages are
+     * exchanged.
+     */
     bool IsInboundConn() const {
         return m_conn_type == ConnectionType::INBOUND;
     }
 
+    /* Whether we send addr messages over this connection */
+    bool RelayAddrsWithConn() const {
+        return m_conn_type != ConnectionType::BLOCK_RELAY;
+    }
+
+    /* Whether we attempt to advertise our own address to the peer when
+     * processing their VERSION message */
+    bool AdvertiseAddressToConn() const {
```

Comment:
I really dislike these functions being inside `CNode` (same as https://github.com/bitcoin/bitcoin/pull/19316#discussion_r464526492). This is very specific net_processing logic, rather than a general property of the CNode. Having the logic inside net.h rather than at the point it's needed makes it less readable.

Reply:
yeah, I agree. this is why I initially opted for having `m_conn_type` as a public member var, but feedback on #19316 led me to switching to private. anyway, given the current state, its a bit unfortunate we don't get the additional guarantee of a switch, but I agree with you & incorporated this feedback so it can be more understandable from the call site.